### ISSUE TITLE
Fails to choose correct GPU

### ISSUE THREAD

**Issue by gudvinr** [Plagman/gamescope#193]
I have multiple GPUs in my system:
```
00:02.0 Display controller: Intel Corporation HD Graphics
01:00.0 VGA compatible controller: Advanced Micro Devices, Inc.
...
/dev/dri/by-path/pci-0000:00:02.0-card -> ../card1
/dev/dri/by-path/pci-0000:01:00.0-card -> ../card0
```

I am using KDE with X11 session and Xorg loading driver for AMD GPU:
```
[    36.724] (II) LoadModule: "amdgpu"
[    36.725] (II) Loading /usr/lib/xorg/modules/drivers/amdgpu_drv.so
[    36.729] (II) Module amdgpu: vendor="X.Org Foundation"
[    36.729]    compiled for 1.20.8, module version = 19.1.0
[    36.729]    Module class: X.Org Video Driver
[    36.729]    ABI class: X.Org Video Driver, version 24.1
```

`$ glxinfo` also recognizes primary GPU correctly:
```
Extended renderer info (GLX_MESA_query_renderer):
    Vendor: AMD (0x1002)
    Device: AMD VEGAM (DRM 3.40.0, 5.10.34-1-lts, LLVM 11.1.0) (0x694e)
    Version: 21.0.3
    Accelerated: yes
    Video memory: 4096MB
```

As well as `$ vulkaninfo`:
```
GPU0:
VkPhysicalDeviceProperties:
---------------------------
        apiVersion     = 4202641 (1.2.145)
        driverVersion  = 88080387 (0x5400003)
        vendorID       = 0x1002
        deviceType     = PHYSICAL_DEVICE_TYPE_DISCRETE_GPU
        deviceName     = AMD RADV VEGAM (ACO)
```

`$ xrandr --listproviders` only shows single rendering device:
```
Providers: number : 1
Provider 0: id: 0x5b cap: 0xf, Source Output, Sink Output, Source Offload, Sink Offload crtcs: 6 outputs: 6 associated providers: 0 name:Unknown AMD Radeon GPU @ pci:0000:01:00.0
```

In fact, Intel GPU has no access to hardware outputs at all and only can be used for quicksync.  
But `gamescope` tries to run with wrong GPU anyway:
```
00:00:00.044 [render/gles2/renderer.c:674] Using OpenGL ES 3.2 Mesa 21.0.3
00:00:00.044 [render/gles2/renderer.c:675] GL vendor: Intel
00:00:00.044 [render/gles2/renderer.c:676] GL renderer: Mesa Intel(R) HD Graphics
```

## Proposet solution

1. Determine "correct" GPU and use the one OS is using (that also might imply support for PRIME or stuff like that)
2. Allow to select GPU. It is not ideal solution and rather looks like workaround but it could help at least to run things now until proper fix implemented

**Comment by emersion** [comment]
The latest commit has a fix for multi-GPU. Can you try again?

**Comment by gudvinr** [comment]
You mean 3.9.2? Thanks for the notice, I'll try once AUR package get an update.

**Comment by emersion** [comment]
Ah yeah the fixes are included in 3.9.2.

**Comment by gudvinr** [comment]
It seems like it's working with both opengl and vulkan applications:

```
$ gamescope -- glxgears
No CAP_SYS_NICE, falling back to regular-priority compute and threads.
Performance will be affected.
vulkan: selecting physical device 'AMD RADV VEGAM'
vulkan: physical device does not support DRM format modifiers
vulkan: supported DRM formats for sampling usage:
vulkan:   0x34325258
vulkan:   0x34325241
vulkan:   0x3231564E
wlserver: [backend/noop/backend.c:59] Creating noop backend
wlserver: Running compositor on wayland display 'gamescope-0'
wlserver: [backend/noop/backend.c:16] Starting noop backend
wlserver: [xwayland/sockets.c:63] Failed to bind socket @/tmp/.X11-unix/X0: Address already in use
wlserver: [xwayland/server.c:84] Starting Xwayland on :1
glamor: No eglstream capable devices found
```

```
$ gamescope -- vkcube
No CAP_SYS_NICE, falling back to regular-priority compute and threads.
Performance will be affected.
vulkan: selecting physical device 'AMD RADV VEGAM'
vulkan: physical device does not support DRM format modifiers
vulkan: supported DRM formats for sampling usage:
vulkan:   0x34325258
vulkan:   0x34325241
vulkan:   0x3231564E
wlserver: [backend/noop/backend.c:59] Creating noop backend
wlserver: Running compositor on wayland display 'gamescope-0'
wlserver: [backend/noop/backend.c:16] Starting noop backend
wlserver: [xwayland/sockets.c:63] Failed to bind socket @/tmp/.X11-unix/X0: Address already in use
wlserver: [xwayland/server.c:84] Starting Xwayland on :1
glamor: No eglstream capable devices found
```
